basic/src/test/java/org/jboss/hal/testsuite/test/configuration/remoting/endpointconfiguration/EndpointConfigurationTestCaseAbstract.java

```diff
@@ -0,0 +1,83 @@
+package org.jboss.hal.testsuite.test.configuration.remoting.endpointconfiguration;
+
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.page.config.RemotingSubsystemPage;
+import org.jboss.hal.testsuite.util.ConfigChecker;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+
+public abstract class EndpointConfigurationTestCaseAbstract {
+
+    @Page
+    protected RemotingSubsystemPage page;
+
+    @Drone
+    protected WebDriver browser;
+
+    protected static final Address
+            REMOTING_SUBSYSTEM_ADDRESS = Address.subsystem("remoting"),
+            ENDPOINT_CONFIGURATION_ADDRESS = REMOTING_SUBSYSTEM_ADDRESS.and("configuration", "endpoint");
+
+    protected static OnlineManagementClient client;
+    protected static Administration administration;
+    protected static Operations operations;
+
+    @BeforeClass
+    public static void beforeClass() {
+        client = ManagementClientProvider.createOnlineManagementClient();
+        administration = new Administration(client);
+        operations = new Operations(client);
+    }
+
+    @Before
+    public void initPage() {
+        page.navigate();
+    }
+
+    @AfterClass
+    public static void afterClass() throws InterruptedException, TimeoutException, IOException {
+        try {
+            administration.reloadIfRequired();
+        } finally {
+            client.close();
+        }
+    }
+
+    protected void enterTextAndVerify(String inputField, String value) throws Exception {
```

Comment:
Just a note: I started writing tests like these, but later it came out better for me to create as independent tests as possible... IMHO abstract methods like these make adapting new APIs and procedures more complicated.